dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -473,12 +465,29 @@ public CloseableClient build() {
                     connectionManager,
                     // Use a shorter check duration than idle connection timeout duration in order to avoid allowing
                     // stale connections to race the server-side timeout.
-                    Duration.ofMillis(Math.min(idleConnectionTimeoutMillis, 5_000)),
-                    Duration.ofMillis(idleConnectionTimeoutMillis));
+                    Duration.ofMillis(Math.min(idleConnectionTimeout.toMilliseconds(), 5_000)),
+                    Duration.ofMillis(idleConnectionTimeout.toMilliseconds()));
             return CloseableClient.wrap(apacheClient, name, connectionManager, connectionEvictorFuture, conf, executor);
         }
     }
 
+    private static Timeout getSocketTimeout(ClientConfiguration conf) {
+        long socketTimeoutMillis = conf.readTimeout().toMillis();
+        if (conf.readTimeout().toMillis() != conf.writeTimeout().toMillis()) {
+            log.warn(
+                    "Read and write timeouts do not match, The value of the readTimeout {} will be used and write "
+                            + "timeout {} will be ignored.",
+                    SafeArg.of("readTimeout", conf.readTimeout()),
+                    SafeArg.of("writeTimeout", conf.writeTimeout()));
+        }
+        if (socketTimeoutMillis == 0) {
+            // https://issues.apache.org/jira/browse/HTTPCLIENT-2099
+            log.debug("Working around HTTPCLIENT-2099 by using a 1 day socket timeout instead of zero (unlimited)");
+            socketTimeoutMillis = Duration.ofDays(1).toMillis();
+        }
```

Comment:
these 5 lines are the only real change, the rest is shuffling away from numbers to proper Timeout values

Reply:
no longer the case, this became a large change which will have a positive impact in several scenarios, particularly for the timelock case.